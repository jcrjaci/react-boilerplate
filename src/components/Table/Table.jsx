import React from 'react';
import PropTypes from 'prop-types';
import TableHeader from '../TableHeader/TableHeader';
import TableBody from '../TableBody/TableBody';

const Table = ({ data, headers, children }) => (
  <div className="table-container">
    <TableHeader headers={headers} />
    <TableBody data={data} />
    {children}
  </div>
);

Table.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  headers: PropTypes.arrayOf(PropTypes.string).isRequired,
  children: PropTypes.node,
};

Table.defaultProps = {
  children: null,
};

export default Table;
