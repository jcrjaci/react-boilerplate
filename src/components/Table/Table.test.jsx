import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Table from './Table';

configure({ adapter: new Adapter() });

describe('<Table />', () => {
  const headers = ['header1', 'header2'];
  const data = [{ id: 1, name: 'bitcoin' }, { id: 2, name: 'litecoin' }];
  const wrapper = shallow(<Table headers={headers} data={data} />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass same props to TableHeader component', () => {
    const tableHeader = wrapper.find('TableHeader');

    expect(tableHeader).toHaveLength(1);
    expect(tableHeader.props().headers).toEqual(headers);
  });

  it('should pass same props to TableBody component', () => {
    const tableBody = wrapper.find('TableBody');

    expect(tableBody).toHaveLength(1);
    expect(tableBody.props().data).toEqual(data);
  });

  it('should render children', () => {
    const children = <div className="children">children</div>;
    wrapper.setProps({ children });

    expect(wrapper.find('.children')).toHaveLength(1);
  });
});
