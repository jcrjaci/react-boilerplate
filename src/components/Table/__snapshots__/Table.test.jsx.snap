// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render correctly 1`] = `
<div
  className="table-container"
>
  <TableHeader
    headers={
      Array [
        "header1",
        "header2",
      ]
    }
  />
  <TableBody
    data={
      Array [
        Object {
          "id": 1,
          "name": "bitcoin",
        },
        Object {
          "id": 2,
          "name": "litecoin",
        },
      ]
    }
  />
</div>
`;
